### ISSUE TITLE
Non-editable install fails on MacOS, because sysconfig.get_platform() is misparsed

### ISSUE THREAD

**Issue by manopapad** [nv-legate/legate.core#648]
The install finishes right after all the files have been copied, with:

```
    File "/Users/mpapadakis/opt/miniconda3/envs/legate/lib/python3.9/site-packages/wheel/bdist_wheel.py", line 281, in get_tag
      plat_name = get_platform(self.bdist_dir)
    File "/Users/mpapadakis/opt/miniconda3/envs/legate/lib/python3.9/site-packages/wheel/bdist_wheel.py", line 52, in get_platform
      result = calculate_macosx_platform_tag(archive_root, result)
    File "/Users/mpapadakis/opt/miniconda3/envs/legate/lib/python3.9/site-packages/wheel/macosx_libfile.py", line 399, in calculate_macosx_platform_tag
      base_version = tuple(int(x) for x in base_version.split("."))
    File "/Users/mpapadakis/opt/miniconda3/envs/legate/lib/python3.9/site-packages/wheel/macosx_libfile.py", line 399, in <genexpr>
      base_version = tuple(int(x) for x in base_version.split("."))
  ValueError: invalid literal for int() with base 10: ''
```

I added some printfs around the failing operation (see them in context below), and it looks like at the final stage the `sysconfig.get_platform()` call returns "macosx--x86_64", whereas in multiple occasions previously it had returned "macosx-10.9-x86_64".

Full build log: https://gist.github.com/manopapad/57833f1e4a2932f2f3b4db72703bafb1

**Comment by bryevdv** [comment]
This is really weird, for sure. The errors are not originating in any of our code AFAICT. Can you provide specific env configuration I can try to reproduce this locally with? @manopapad

**Comment by manopapad** [comment]
I built from scratch as follows:

```
~/legate.core> ./scripts/generate-conda-envs.py --python 3.9 --ctk none --os osx --compilers --openmpi --no-ucx
--- generating: environment-test-osx-py3.9-compilers-openmpi.yaml
~/legate.core> conda env create -n py39 -f environment-test-osx-py3.9-compilers-openmpi.yaml
~/legate.core> conda activate py39
~/legate.core> ./install.py --verbose --no-clean --no-build-isolation --legion-src-dir ../legion --debug
```

Full output (edited-out some typos etc.): https://gist.github.com/manopapad/1508f8baf870eabea0f46e476d0e476b

**Comment by bryevdv** [comment]
Noting that a simpler `./install.py --verbose --clean --debug` also repos w/o needing to fetch legion source. 

I am at a bit of a loss, just running `get_platform()` in the REPL works fine, of course:
```
py39 ❯ python
Python 3.9.16 | packaged by conda-forge | (main, Feb  1 2023, 21:42:20)
[Clang 14.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import sysconfig
>>> sysconfig.get_platform()
'macosx-10.9-x86_64'
```
This bad value is showing up in the paths to the directories that `skbuild` creates. e.g.
```
 -- Installing: /Users/bryan/work/legate.core/_skbuild/macosx--x86_64-3.9/cmake-install/bin/legion_python
``` 
Is it possible this is an issue stemming from `skbuild` somehow? cc @trxcllnt in case anything obvious comes to mind

**Comment by bryevdv** [comment]
Noting that `skbuild` APIs used in their `setup` wrapper also return expected values from the REPL
```
>>> import skbuild.constants as skc
>>> skc.skbuild_plat_name()
'macosx-13.0-x86_64'
```

**Comment by trxcllnt** [comment]
Seems like a setuptools issue: https://github.com/grpc/grpc/issues/31800#issuecomment-1338498791

**Comment by bryevdv** [comment]
That issue seems to suggest the issue lies with the vendored `distutils` inside `setuptools`, but at least on my system (where the build fails), the API in questions seems to give the expected answer:
```
>>> setuptools.distutils.util.get_platform()
'macosx-10.9-x86_64'
```
This is with latest setuptools 67.6.1 I also tried with older setuptools 61.0.0, since this did work in the past. 

It may end up being setuptools but the only concrete thread I see to pull on is that problems are first *evident* in the skbuild directory paths, so I am inclined to open an issue there to start. I am going to see if changing skbuild versions has any effect first, though.

**Comment by bryevdv** [comment]
That said, this mac has acos 13.3 (22E252) installed, so the version above is technically wrong, but at least something parseable is there...

**Comment by bryevdv** [comment]
ref: https://github.com/scikit-build/scikit-build/issues/911

Will see if skbuild folks can help us pull on this thread.

**Comment by manopapad** [comment]
> The full build still ultimately fails on OSX, but now for other reasons.

What's the new error? Should we track that on a different issue?

**Comment by bryevdv** [comment]
@manopapad the new issue is that on OSX, cmake variables `Legion_USE_CUDA` etc are... unset? null? They show up as nothing at all in `install_info.py` rather then `OFF`, which results in a syntax error.

This diff "fixes" the issue, but I was going to see if there is something better. 
```diff
diff --git a/legate/install_info.py.in b/legate/install_info.py.in
index 0d20cdf..7ccf235 100644
--- a/legate/install_info.py.in
+++ b/legate/install_info.py.in
@@ -51,6 +51,6 @@ build_type: str = "@CMAKE_BUILD_TYPE@"

 ON, OFF = True, False

-use_cuda: bool = @Legion_USE_CUDA@
+use_cuda: bool = bool(@Legion_USE_CUDA@)

-use_openmp: bool = @Legion_USE_OpenMP@
\ No newline at end of file
+use_openmp: bool = bool(@Legion_USE_OpenMP@)
\ No newline at end of file
```

Additionally, there seems to be some intermittent issue involving the jupyter extension at the end as well.

**Comment by bryevdv** [comment]
Here is the jupyter error, it shows up on some clean builds but not others. I don't think it affects the rest of the install, though
```
Executing: " cmake --install /Users/bryan/work/legate.core/_skbuild/macosx-13.0-x86_64-3.9/cmake-build/_deps/legion-build/jupyter_notebook --prefix /Users/bryan/anaconda/envs/py39 " with  {}
CMake Error: Error processing file: /Users/bryan/work/legate.core/_skbuild/macosx-13.0-x86_64-3.9/cmake-build/_deps/legion-build/jupyter_notebook/cmake_install.cmake
Traceback (most recent call last):
  File "/Users/bryan/work/legate.core/./install.py", line 795, in <module>
    driver()
  File "/Users/bryan/work/legate.core/./install.py", line 791, in driver
    install(unknown=unknown, **vars(args))
  File "/Users/bryan/work/legate.core/./install.py", line 498, in install
    install_legion_jupyter_notebook(
  File "/Users/bryan/work/legate.core/./install.py", line 223, in install_legion_jupyter_notebook
    execute_command(
  File "/Users/bryan/work/legate.core/./install.py", line 74, in execute_command
    subprocess.check_call(args, **kwargs)
  File "/Users/bryan/anaconda/envs/py39/lib/python3.9/subprocess.py", line 373, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '--install', '/Users/bryan/work/legate.core/_skbuild/macosx-13.0-x86_64-3.9/cmake-build/_deps/legion-build/jupyter_notebook', '--prefix', '/Users/bryan/anaconda/envs/py39']' returned non-zero exit status 1.
```

**Comment by manopapad** [comment]
> Here is the jupyter error, it shows up on some clean builds but not others.

CMake doesn't seem to give us much info on the cause of the failure. @trxcllnt any idea what could be going wrong here?

And also, do you know why the  `@Legion_USE_CUDA@` substitutions could be turning up empty, from https://github.com/nv-legate/legate.core/issues/648#issuecomment-1502132248?

**Comment by bryevdv** [comment]
@manopapad @trxcllnt OK I think maybe we tracked down the skbuild issue. Activating the conda env sets this `CMAKE_ARGS`

> 
> py39 ❯ echo $CMAKE_ARGS
> -DCMAKE_AR=/Users/bryan/anaconda/envs/py39/bin/x86_64-apple-darwin13.4.0-ar -DCMAKE_CXX_COMPILER_AR=/Users/bryan/anaconda/envs/py39/bin/x86_64-apple-darwin13.4.0-ar -DCMAKE_C_COMPILER_AR=/Users/bryan/anaconda/envs/py39/bin/x86_64-apple-darwin13.4.0-ar -DCMAKE_RANLIB=/Users/bryan/anaconda/envs/py39/bin/x86_64-apple-darwin13.4.0-ranlib -DCMAKE_CXX_COMPILER_RANLIB=/Users/bryan/anaconda/envs/py39/bin/x86_64-apple-darwin13.4.0-ranlib -DCMAKE_C_COMPILER_RANLIB=/Users/bryan/anaconda/envs/py39/bin/x86_64-apple-darwin13.4.0-ranlib -DCMAKE_LINKER=/Users/bryan/anaconda/envs/py39/bin/x86_64-apple-darwin13.4.0-ld -DCMAKE_STRIP=/Users/bryan/anaconda/envs/py39/bin/x86_64-apple-darwin13.4.0-strip -DCMAKE_INSTALL_NAME_TOOL=/Users/bryan/anaconda/envs/py39/bin/x86_64-apple-darwin13.4.0-install_name_tool -DCMAKE_LIBTOOL=/Users/bryan/anaconda/envs/py39/bin/x86_64-apple-darwin13.4.0-libtool ***-DCMAKE_OSX_DEPLOYMENT_TARGET=*** -DCMAKE_BUILD_TYPE=Release -DCMAKE_OSX_SYSROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk

Note the `-DCMAKE_OSX_DEPLOYMENT_TARGET=` near the end. If I delete `CMAKE_ARGS` entirely, then the build completes, usind the correct macos version (reflected in the paths). I am not sure if removing all those cmake args entirely causes any other issues, though. 

TLDR; I guess this comes down to a conda issue?

**Comment by trxcllnt** [comment]
The conda compilers strike again!

**Comment by manopapad** [comment]
Yes, it looks like this is set during activation of the `clang` package https://github.com/conda-forge/clang-compiler-activation-feedstock/blob/main/recipe/activate-clang.sh#L136.

I was able to build (non-editable) on an environment that doesn't pull clang at all (using the MacOS system compilers). It's on my TODO list to drop compilers altogether from our developer environments, so that will mostly take care of that. We also pull `clang` explicitly, presumably to enable `clang-tools`, but that doesn't seem to actually be necessary (`clang-tools` only depends on `libclang`, and I was able to run the pre-commit using an env with `clang-tools` but w/o `clang`).

**Comment by bryevdv** [comment]
A fix was submitted on conda-forge: https://github.com/conda-forge/clang-compiler-activation-feedstock/pull/102

@manopapad can this be closed, then? I don't think there are any actions on our end.

**Comment by bryevdv** [comment]
@henryiii is there a typical ETA on when new packages are available? I can only seem to get build number 0 still.

**Comment by henryiii** [comment]
I think I’ve heard six hours for mirrors. Under 24 for sure.

**Comment by bryevdv** [comment]
@henryiii the change seems to be working

**Comment by bryevdv** [comment]
AFAIK this can be closed at this time
